ui/vendor/_directional.scss

```diff
@@ -0,0 +1,63 @@
+// directional-scss | Author: Tyson Matanich (http://matanich.com), 2013 | License: MIT
+$dir: ltr !default;
+
+// Default $dir if not valid
+@if $dir != ltr and $dir != rtl {
+    $dir: ltr;
+}
+
+@function if-ltr($if, $else: null) {
+  @if $dir != rtl {
+    @return $if;
+  }
+  @else {
+    @return $else;
+  }
+}
+
+@function if-rtl($if, $else: null) {
+  @return if-ltr($else, $if);
+}
+
+$left: if-ltr(left, right);
+$right: if-ltr(right, left);
```

Comment:
The last code sample you sent for the `.slds-input` is very good, the problem of adding `slds-p-left--large` and having overlaps is the same currently without the `rtl` support.

I think the directional stuff can be separated from the regular code, maybe on `flavors/directional/_index.scss`, so on the current code we just need to add `-start` and `-end` where we use `-left` and `-right` maybe `-left` and `-right` can be deprecated in future versions. An on the `flavors/directional/_index.scss` we can add only the `rtl` overrides.

```
[dir=rtl] {
   .slds-table--edit_container {
      .slds-table--edit_container-message {
          right: 50%;
          margin-right: (($size-medium / 2) * -1);
      }
   }
}
```